sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkTracerProviderBuilder.java

```diff
@@ -76,19 +115,19 @@ public SdkTracerProviderBuilder setIdGenerator(IdGenerator idGenerator) {
    */
   public SdkTracerProviderBuilder setResource(Resource resource) {
     requireNonNull(resource, "resource");
-    this.resource = resource;
+    this.entityProvider = new SdkEntityProvider(Collections.singletonList(Entity.create(resource)));
```

Comment:
This will need to be much more sophisticated to deal with Resource vs. Entity shenanigans.